src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1159,9 +1159,10 @@
     (add-certname! certname))
   (let [timestamp (to-timestamp time)
         replaced  (sql/update-values :certnames
-                                     ["name=? AND (deactivated<? OR deactivated IS NULL)" certname timestamp]
-                                     {:deactivated nil})]
-    (pos? (first replaced))))
+                                     ["name=? AND deactivated<?" certname timestamp]
```

Comment:
I've confirmed a speedup. See these screenshots:
![selection_003](https://cloud.githubusercontent.com/assets/4952091/7886437/f6feccc2-05de-11e5-8824-125ad573c825.png)
![selection_002](https://cloud.githubusercontent.com/assets/4952091/7886441/fe04a35c-05de-11e5-8ea7-2ac8b4771e4b.png)

the first is before the change, and the second is after. We should be cautious of unintended consequences like you're describing though.